src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(
+    person => person.sex === 'm'
+      && (!century || century === Math.ceil(person.died / 100))
+  );
+
+  const menAge = men.map(person => person.died - person.born);
+
+  const sumMenAge = menAge.reduce((prev, next) => prev + next, 0);
+
+  return sumMenAge / menAge.length;
```

Comment:
Create separate function-helper with this functionality and use it here and in `calculateWomenAverageAge`. Becase you have same code - do not repeat your self